### PR TITLE
 [REPO-5552] more filtering capablities for aspect/type api

### PR DESCRIPTION
**Pull request by dhrn**
Following capability added to aspect/type api

* include properties
* include associations
* INCLUDESUBTYPES
* INCLUDESUBASPECTS
* typo fixed

### CODE REVIEW

**Review by skopf — APPROVED** [review]
So you are making properties an optional feature, right? Makes sense. Sometimes, we just need the basic information about types/aspects to render a UI but we do not need all the details.
If you are already looking at making property definitions optional, you should also look at associations for types. They are typically not required in many cases and should also be optional.

**Review by skopf — CHANGES_REQUESTED** [review]
So you are making properties an optional feature, right? Makes sense. Sometimes, we just need the basic information about types/aspects to render a UI but we do not need all the details.
If you are already looking at making property definitions optional, you should also look at associations for types. They are typically not required in many cases and should also be optional.
But we cannot make them optional in a later version of this API. So we should include this in this first revision.

**Comment by skopf** [comment]
The confluence page does not correctly describe the layout we need for associations. We actually need:
```
id -> getName()
title ->  getTitle()
description -> getDescription()
child -> isChild
protected -> isProtected()
```
plus two JSON child objects for `source`and `target` with these properties  each
```
role -> get[ST]RoleName()
cls -> get[ST]Class()
many -> is[Source|Target]Many()
mandatory -> is[Source|Target]Mandatory()
mandatoryEnforced -> is[Source|Target]MandatoryEnforced()
```

File: src/main/webapp/definitions/alfresco-model.yaml [hunk 1]
```diff
@@ -114,27 +114,43 @@ paths:
...
           required: false
           type: string
         - $ref: '#/parameters/skipCountParam'
         - $ref: '#/parameters/maxItemsParam'
+        - name: include
+          in: query
+          description: |
+            Returns additional information about the aspect. The following optional fields can be requested:
+            * properties
+            * container
```
Comment (gavincornwell) [inline src/main/webapp/definitions/alfresco-model.yaml:144]: We typically use the include mechanism for data that could be expensive to retrieve or make the response very large, properties, associations and mandatoryAspects are good examples of this. 

Personally I would just have boolean properties in the response by default.

File: src/main/webapp/definitions/alfresco-model.yaml [hunk 2]
```diff
@@ -517,3 +563,35 @@ definitions:
...
+  AssociationSource:
+    type: object
+    properties:
+      role:
+        type: string
+      cls:
+        type: string
+      isMandatory:
+        type: boolean
+      many:
```
Comment (gavincornwell) [inline src/main/webapp/definitions/alfresco-model.yaml:594]: Boolean properties should be prefixed with "is" according to the guidelines.

**Review by gavincornwell — CHANGES_REQUESTED** [review]

**Review by gavincornwell — APPROVED** [review]
LGTM

**Review by skopf — APPROVED** [review]
LGTM

**Review by eromano — APPROVED** [review]
LGTM thanks @dhrn

**Pull request merged**